extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -418,6 +427,57 @@ export const popupMessageListener = (request: Request) => {
     };
   };
 
+  const addCustomNetwork = () => {
+    const { customNetwork } = request;
+    const savedNetworks = JSON.parse(
+      localStorage.getItem(NETWORKS_LIST_ID) ||
```

Comment:
thoughts on abstracting this? It seems like we will need to be returning the default_networks anytime the networks_list_id doesn't exist, and to protect us down the road from forgetting network_list_id doesn't necessarily exist in the localstorage. Could be something like:

```
const getSavedNetworks = () => {
    return JSON.parse(
      localStorage.getItem(NETWORKS_LIST_ID) ||
        JSON.stringify(DEFAULT_NETWORKS),
    );
}